api/helpers.go

```diff
@@ -62,30 +62,54 @@ func findEntry(ctx context.Context, root plugin.Entry, segments []string) (plugi
 	return curEntry, nil
 }
 
-func getPathFromRequest(r *http.Request) (string, *errorResponse) {
+func getFullPathFromRequest(r *http.Request) (string, *errorResponse) {
 	paths := r.URL.Query()["path"]
 	if len(paths) != 1 {
 		return "", invalidPathsResponse(paths)
 	}
-	return paths[0], nil
-}
-
-func getEntryFromRequest(ctx context.Context, r *http.Request) (plugin.Entry, string, *errorResponse) {
-	path, errResp := getPathFromRequest(r)
-	if errResp != nil {
-		return nil, "", errResp
-	}
+	path := paths[0]
 
 	if path == "" {
 		panic("path should never be empty")
 	}
 	if !filepath.IsAbs(path) {
-		return nil, "", relativePathResponse(path)
+		return "", relativePathResponse(path)
 	}
 
+	return path, nil
+}
+
+func getWashPathFromFullPath(ctx context.Context, path string) (string, *errorResponse) {
 	mountpoint := ctx.Value(mountpointKey).(string)
 	trimmedPath := strings.TrimPrefix(path, mountpoint)
 	if trimmedPath == path {
+		return path, notCachedPathResponse(path)
+	}
+
+	return trimmedPath, nil
+}
+
+func getWashPathFromRequest(r *http.Request) (string, *errorResponse) {
```

Comment:
I'm guessing you added this to avoid unnecessary API requests for endpoints that do not require an Entry object? If so, a comment explaining that would be nice.